src/Settings/Packages/Rts.hs

```diff
@@ -29,13 +38,67 @@ rtsLibffiLibraryName = do
         (False, False) -> "Cffi"
         (_    , True ) -> "Cffi-6"
 
+rtsLibffiIncludeArgs :: Args
+rtsLibffiIncludeArgs = package libffi ? builder (Ghc CompileCWithGhc) ? do
+  useSystemFfi <- expr $ flag UseSystemFfi
+  ffiIncludeDir <- getSetting FfiIncludeDir
+  mconcat [
+    useSystemFfi ? pure (map ("-I" ++) $ words ffiIncludeDir),
+    -- ffi.h triggers prototype warnings, so disable them here:
+    inputs [ "//Interpreter.c", "//Storage.c", "//Adjustor.c" ] ?
+    arg "-Wno-strict-prototypes" ]
+
 rtsLibffiLibrary :: Way -> Action FilePath
 rtsLibffiLibrary way = do
     name    <- rtsLibffiLibraryName
     suf     <- libsuf way
     rtsPath <- rtsBuildPath
     return $ rtsPath -/- "lib" ++ name ++ suf
 
+-- ref: mk/config.mk.in
+ghcRtsWithLibDw :: Action Bool
+ghcRtsWithLibDw = do
+    goodArch <- anyTargetArch ["i386", "x86_64"]
+    withLibDw <- setting WithLibDw
+    return $ goodArch && withLibDw == "YES"
+
+ccClangBackend :: Action Bool
+ccClangBackend = (== "1") <$> setting CcClangBackend
```

Comment:
Ohh, is `"1"` here equivalent to `"YES"`? Or is this a version number?

Reply:
If this is a version, I suggest we rename this to something like `ccClangBackendVersionEq1` to distinguish it from YES/NO settings.

Reply:
I think it is using "1" to represent "YES", and "0" for "NO".  Can we simply use `GccIsClang` here?

Reply:
See https://github.com/ghc/ghc/blob/master/aclocal.m4#L1250

Reply:
@izgzhen Yes, it looks like we can use `GccIsClang` instead. Perhaps we could even drop `CcClangBackend` altogether?

Reply:
Indeed, I was thinking about doing that